### PR TITLE
ui: use NPM dependencies mirrored to GCS bucket instead of yarn-vendored

### PR DESCRIPTION
**Pull request by sjbarag**
Now that tooling exists to mirror NPM dependencies and rewrite the
relevant yarn.lock entries [1], there's nothing stopping those
dependencies from getting mirrored and being used. ./dev ui mirror-deps
has already been run. Update the lockfiles, switch to online
installation when fetching dependencies with Bazel, and remove both the
"seed_yarn_cache" hack and the pkg/ui/yarn-vendor submodule.

[1] https://github.com/cockroachdb/cockroach/pull/94102

### CODE REVIEW

**Comment by sjbarag** [comment]
Now 100% more docs!

File: WORKSPACE [hunk 1]
```diff
@@ -266,13 +264,9 @@ yarn_install(
 yarn_install(
     name = "npm_eslint_plugin_crdb",
     args = [
-        "--offline",
+        "--pure-lockfile",
```
Comment (rickystewart) [inline WORKSPACE:267]: Can you additionally update the `Makefile` to use `--pure-lockfile` instead of `--offline`?

File: pkg/ui/BUILD.bazel [hunk 2]
```diff
@@ -49,6 +49,7 @@ EOF
 test_suite(
     name = "lint",
     tests = [
+        "//pkg/cmd/mirror/npm:are_lockfiles_updated",
```
Comment (rickystewart) [inline pkg/ui/BUILD.bazel:52]: My understanding is this test will fail in CI if someone forgets to run `dev ui mirror-deps`, right?

**Review by rickystewart — APPROVED** [review]

File: pkg/ui/BUILD.bazel (hunk 2 above)
Comment (sjbarag) [reply pkg/ui/BUILD.bazel:52]: Correct! This shouldn't be an issue for most folks, and should only come into play for devs who add or update dependencies.

File: WORKSPACE (hunk 1 above)
Comment (sjbarag) [reply WORKSPACE:267]: Oh sure!  I forgot we're continuing to maintain that :)

Comment (sjbarag) [reply WORKSPACE:267]: Done! Rebased on master and pushed again :)

**Comment by sjbarag** [comment]
bors r=rickystewart,nathanstilwell

File: pkg/ui/README.md [hunk 3]
```diff
@@ -117,129 +117,87 @@ $ ./dev ui test
...
+   shared dependencies into pkg/ui/node_modules, so **dependencies are managed
+   independently in each sub-project.**
+2. The public registries can be used for initial installation, but have to
+   be rewritten before a PR can merge. The linter will help with this, no
+   need to worry.
+
+### Adding, Removing, or Updating a dependency
+Since there's no shared dependency hoisting and dependencies are managed
+independently for each tree under pkg/ui/workspaces (see above), adding,
+removing, and updating dependencies must happt happen from the root of the sub-
```
Comment (nathanstilwell) [inline pkg/ui/README.md:148]: `must happt happen` typo?

**Review by nathanstilwell — APPROVED** [review]
This day has been a long time coming. Well done @sjbarag. Besides one typo in the README, LGTM

File: pkg/ui/README.md (hunk 3 above)
Comment (sjbarag) [reply pkg/ui/README.md:148]: doh, yes

**Comment by sjbarag** [comment]
bors cancel

**Pull request merged**